src/hooks/useUsersGuildPins.ts

```diff
@@ -19,70 +101,48 @@ const fetchGuildPinsOnChain = async (
     transport: http(),
   })
 
-  const usersGuildPinIdsOnChain: bigint[] = []
-
-  const balance = await publicClient.readContract({
-    abi: GUILD_PIN_CONTRACTS[chain].abi,
-    address: GUILD_PIN_CONTRACTS[chain].address,
-    functionName: "balanceOf",
-    args: [address],
-  })
-
-  for (let i = 0; i < balance; i++) {
-    const newTokenId = await publicClient.readContract({
+  let balance = null
+  try {
+    balance = await publicClient.readContract({
       abi: GUILD_PIN_CONTRACTS[chain].abi,
       address: GUILD_PIN_CONTRACTS[chain].address,
-      functionName: "tokenOfOwnerByIndex",
-      args: [address, BigInt(i)],
+      functionName: "balanceOf",
+      args: [address],
     })
-
-    if (newTokenId) usersGuildPinIdsOnChain.push(newTokenId)
+  } catch (e) {
+    return { usersPinsMetadataJSONs: [], errors: [{ error: e, status: "failure" }] }
```

Comment:
Please type this (even just by adding type casts), so we get a nice clean return type for the `fetchGuildPinsOnChain` function!